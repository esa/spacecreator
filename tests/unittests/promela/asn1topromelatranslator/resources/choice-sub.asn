WithComponents DEFINITIONS ::= BEGIN

MyChoice ::= CHOICE
{
    firstOption INTEGER (0..2),
    secondOption INTEGER(0..3),
    thirdOption INTEGER(0..4)
}

ConcreteChoice ::= MyChoice (WITH COMPONENTS { firstOption, thirdOption })

END