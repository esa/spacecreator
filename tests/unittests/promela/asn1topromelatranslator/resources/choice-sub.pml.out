#define ConcreteChoice_firstOption_PRESENT 1
#define ConcreteChoice_thirdOption_PRESENT 2
inline ConcreteChoice_firstOption_generate_value(ConcreteChoice_firstOption_gv)
{
    int ConcreteChoice_firstOption_tmp;
    if
    ::  atomic {
        select (ConcreteChoice_firstOption_tmp : 0 .. 2);
    }
    fi;
    ConcreteChoice_firstOption_gv = ConcreteChoice_firstOption_tmp;
}
inline ConcreteChoice_thirdOption_generate_value(ConcreteChoice_thirdOption_gv)
{
    int ConcreteChoice_thirdOption_tmp;
    if
    ::  atomic {
        select (ConcreteChoice_thirdOption_tmp : 0 .. 4);
    }
    fi;
    ConcreteChoice_thirdOption_gv = ConcreteChoice_thirdOption_tmp;
}
inline ConcreteChoice_generate_value(ConcreteChoice_gv)
{
    if
    ::  true;
        ConcreteChoice_firstOption_generate_value(ConcreteChoice_gv.firstOption);
        ConcreteChoice_gv.selection = ConcreteChoice_firstOption_PRESENT;
    ::  true;
        ConcreteChoice_thirdOption_generate_value(ConcreteChoice_gv.thirdOption);
        ConcreteChoice_gv.selection = ConcreteChoice_thirdOption_PRESENT;
    fi;
}
